src/testing/builder/CategoriesMapModelBuilder.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+
+import { CategoriesMapModel, CategoryModel } from '@integreat-app/integreat-api-client'
+import moment from 'moment-timezone'
+import type { FileCacheStateType, LanguageResourceCacheStateType } from '../../modules/app/StateType'
+import seedrandom from 'seedrandom'
+import hashUrl from '../../modules/endpoint/hashUrl'
+
+const DEFAULT_MAX_WIDTH = 3
+const DEFAULT_DEPTH = 2
+const MAX_PREDICTABLE_VALUE = 6
+
+class CategoriesMapModelBuilder {
+  _depth: number
+  _maxWidth: number
+
+  constructor (maxWidth: number = DEFAULT_MAX_WIDTH, depth: number = DEFAULT_DEPTH) {
+    this._maxWidth = maxWidth
+    this._depth = depth
+  }
+
+  predictableNumber (index: number, max: number = MAX_PREDICTABLE_VALUE): number {
+    return seedrandom(`${index}-seed`)() * max
+  }
+
+  createResource (url: string, index: number, lastUpdate: moment): FileCacheStateType {
+    const hash = hashUrl(url)
+    return {
+      [url]: {
+        filePath: `path/to/documentDir/resource-cache/v1/some-city/files/${hash}.png`,
+        lastUpdate: moment(lastUpdate).add(this.predictableNumber(index), 'days'),
+        hash
+      }
+    }
+  }
+
+  addChildren (
+    category: CategoryModel,
+    categories: Array<CategoryModel>,
+    resourceCache: LanguageResourceCacheStateType,
+    depth: number) {
+    if (depth === this._depth) {
+      return
+    }
+
+    for (let i = 0; i < this._maxWidth; i++) {
+      const path = `${category.path}/category_${depth}-${i}`
+      const lastUpdate = moment('2017-11-18T19:30:00.000Z', moment.ISO_8601)
+
+      const newChild = new CategoryModel({
+        id: 0,
```

Comment:
Maybe using different ids would make it easier to distinguish categories sometimes